### ISSUE TITLE
How do I get Cognito Access token using Federated SignIn?

### ISSUE THREAD

**Issue by michelmob** [aws/aws-amplify#1073]
Do you want to request a feature or report a bug?

> report

What is the current behavior?

> I'm receiving the CognitoIdentityCredentials and not the cognito access token

What is the expected behavior?

> Mybe, but I need to get the access token.

Which versions of Amplify, and which browser / OS are affected by this issue? Did this work in previous versions?

> I don't think so.

You can turn on the debug mode to provide more info for us by setting ```window.LOG_LEVEL = 'DEBUG';``` in your app.

> More info
> https://stackoverflow.com/q/50959946/319206

**Comment by haverchuck** [comment]
@michelmob 

Thanks for reaching out!

How are you trying to retrieve the token after login?

Have you tried this?

import { Cache } from 'aws-amplify';
Cache.getItem('federatedInfo').token;

(Answer also posted to SO)

**Comment by haverchuck** [comment]
Hi @michelmob ,

The github wiki FAQ has an [entry on this topic](https://github.com/aws/aws-amplify/wiki/FAQ#how-can-i-get-jwt-token-when-using-amplify-to-get-federated-users-login)

tldr: 

If you are using federated identities, then you will only have access to the token that is received from the provider (Facebook, Amazon, etc), not from Cognito.  You can access this token by importing Amplify's Cache module and using `Cache.getItem('federatedInfo')`.

If you want a token actually from Cognito, you can go the [Hoisted UI](https://aws.github.io/aws-amplify/media/authentication_guide#using-amazon-cognito-hosted-ui) route.

 Closing this issue for now - please let us know if this doesn't help!

**Comment by michelmob** [comment]
OK, thank you so much.
